pyanaconda/modules/storage/disk_initialization/configuration.py

```diff
@@ -122,16 +125,22 @@ def can_initialize(self, storage, disk):
         :param disk: an instance of the disk we want to format
         :return: True or False
         """
+        log.debug("Can %s be initialized?", disk.name)
+
         # Skip protected and readonly disks.
         if disk.protected:
+            log.debug("A protected disk cannot be initialized.")
             return False
 
         # Initialize disks with unrecognized formatting.
         if self.format_unrecognized and disk.format.type is None:
+            log.debug("A disk with unrecognized formatting can be initialized.")
```

Comment:
Just wondering (I'm afraid I don see deeply enough into use cases for this) but is this gated by user selecting the disk for installation, right ? 

Just thinking that otherwise there could be a partially corrupted disk connected to the machine or just a disk with totally custom data (IIRC some encryption systems don't have any headers on media on purpose) being written to, with possibly destructive results.

Reply:
> Just wondering (I'm afraid I don see deeply enough into use cases for this) but is this gated by user selecting the disk for installation, right ?

Of course.